### DISCUSSION TITLE
How do u test gokit endpoint

### DISCUSSION THREAD

**Discussion by chaudharisuresh997** [go-kit/kit#1142] [Q&A]
Can u please post and test where u are testing an endpoint of gokit.

**Comment by peterbourgon** [comment]
An endpoint is just a function like any other, so you can create a mock service, generate endpoints for it, and test the inputs and outputs of those functions, just like any other unit test. For a more comprehensive approach, [here is an example of an end-to-end "wireup" test](https://github.com/go-kit/kit/blob/master/examples/addsvc/cmd/addsvc/wiring_test.go). 

Is there something more specific you want to test?

**Reply by vpiyush** [reply]
The link seems to be broken can you please fix it.

**Reply by peterbourgon** [reply]
See https://github.com/go-kit/examples

**Reply by da-revo** [reply]
Here is the specific link:
https://github.com/go-kit/examples/blob/master/addsvc/cmd/addsvc/wiring_test.go
